docs/DeveloperGuide.md

```diff
@@ -1,34 +1,851 @@
+
+#### Design Considerations
+
+* **Alternative 1 (current choice)**: Saves the user input to the file
+    * Pros: Easy to implement as code from `Parser` can be reused
+    * Cons: Increases coupling as it depends on the `Parser` class 
+    to make sense of the data
+* **Alternative 2**: Parses input into a different format 
+storing the different modules, time expected, time spent and tasks
+    * Pros: Independent of how `Parser` takes in user input
+    * Cons: Requires additional work to parse data into the required
+    storage format
+
+
+### 4.2 Add Module <a name="4.2"></a>
+
+#### Current Implementation
+
+The add module feature will implement the checks of the module code input by the user through the following operations:
+
+`checkIfModuleValid()` – Checks if the module code starts with 2 - 3 characters, followed by 4 digits, followed by an 
+                         optional character.
+
+`checkIfModuleExist()` – Checks if the module already exists in the list of modules
+
+If the module code is valid and does not exist, the add module feature will proceed to create the module and adds it to 
+the list of modules.  
+
+> **_NOTE:_**  A module is identfied by its module code
+
+The following activity diagram summarizes what happens when a user executes `addmod CS2113T` command for the first time.  
+
+![addmod](diagrams/addmod%20activity%20diagram.png)
+
+Given below is a detailed explanation on how the add module mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time and hence, the list of modules is empty. 
+        The user types in a command: `addmod CS2113T`. 
+
+Step 2. The parser class makes sense of the user input and calls the `addMod()` function in ModuleList class.
+
+Step 3. Within the `addMod()` function, it will first check if the module code is valid by calling 
+        `checkIfModuleValid()` function. The `checkIfModuleValid()` function will return true if the module code is 
+        valid, and return false otherwise. If `checkIfModuleValid()` function returns true, proceed to step 4.
+
+Step 4. Within the addmod function, it will then check if the module already exists in the list of modules by 
+        calling `checkIfModuleExist()` function. The `checkIfModuleExist()` function will return true if the module 
+        already exists in the modlist, and false otherwise. If `checkIfModuleValid()` function returns true, 
+        proceed to step 5.  
+
+Step 5. A new module with module code "CS2113T” is created by calling the constructor of Module class.
+
+Step 6. The newly created module is added to the list of modules.
+
+Step 7. The `printAdd()` function of the Ui class is called, displaying “CS2113T is added.” to the user.
+
+Step 8. The newly created module is saved to storage.  
+  
+
+The following sequence diagram shows how the `addmod CS2113T` command works.
+
+![Addmodseq](diagrams/Addmodseq.png)  
```

Comment:
Is there a missing return arrow here as stated in Step 4?
![review 9](https://user-images.githubusercontent.com/57140494/97388176-836d0d80-1912-11eb-9138-561185ecabfc.png)